lib/turkish_stemmer.rb

```diff
@@ -36,19 +36,16 @@ def stem(original_word)
     # Preprocess
     return original_word if !proceed_to_stem?(original_word)
 
-    word = original_word.dup
-
     # Process
-    stems = []
-    stems << nominal_verbs_suffix_machine(word)
-    stems << original_word
-    stems.flatten!.uniq!
-    stems << stems.map { |word| noun_suffix_machine(word) }
-    stems << original_word
-    stems.flatten!.uniq!
-    stems << stems.map { |word| derivational_suffix_machine(word) }
-
-    # Postprocess
+    # set of stem candidates
+    stems = [original_word, *nominal_verbs_suffix_machine(original_word.dup)]
```

Comment:
While I agree that this code can be simplified, I'm not sure I agree with this approach:

- I like the step by step approach of the original. I think it enhances readability.
- I agree that calls to `flatten!` can be replaced, though I prefer the `+` array operator to splatting.
- Calling `uniq!` before mapping the array to the `noun_suffix_machine` method makes sense.
- I think calls to `&method` is a bit of a code smell (and I think hinder performance a bit as well, see [here](https://blog.codeminer42.com/towards-minimal-idiomatic-and-performant-ruby-code-f3fc6aed3c94#.x1x10jnkr)).

As a final thought I don't like the need for using `dup` on the original before calling methods that are not explicitly mutating. Ideally I think these methods should call `dup` themselves on their arguments. Otherwise I think they should have a bang `!` operator in their names.